### PR TITLE
Use a new utility method to get data from the XMLRPC response

### LINKED ISSUE #164: Crash: NumberFormatException: Invalid long in PostXMLRPCClient.postResponseObjectToPostModel
I think we have a `Long.valueOf` somewhere we should have a `Double.valueOf`:

```
09-06 04:53:29.004 3180-3180/? E/AndroidRuntime: FATAL EXCEPTION: main
                                                 Process: org.wordpress.android.fluxc.example, PID: 3180
                                                 java.lang.NumberFormatException: Invalid long: "65.9667"
                                                     at java.lang.Long.invalidLong(Long.java:124)
                                                     at java.lang.Long.parse(Long.java:363)
                                                     at java.lang.Long.parseLong(Long.java:353)
                                                     at java.lang.Long.parseLong(Long.java:321)
                                                     at java.lang.Long.valueOf(Long.java:511)
                                                     at org.wordpress.android.fluxc.network.xmlrpc.post.PostXMLRPCClient.postResponseObjectToPostModel(PostXMLRPCClient.java:307)
                                                     at org.wordpress.android.fluxc.network.xmlrpc.post.PostXMLRPCClient.postsResponseToPostsModel(PostXMLRPCClient.java:234)
                                                     at org.wordpress.android.fluxc.network.xmlrpc.post.PostXMLRPCClient.access$400(PostXMLRPCClient.java:43)
                                                     at org.wordpress.android.fluxc.network.xmlrpc.post.PostXMLRPCClient$3.onResponse(PostXMLRPCClient.java:112)
                                                     at org.wordpress.android.fluxc.network.xmlrpc.post.PostXMLRPCClient$3.onResponse(PostXMLRPCClient.java:104)
                                                     at org.wordpress.android.fluxc.network.xmlrpc.XMLRPCRequest.deliverResponse(XMLRPCRequest.java:51)
                                                     at com.android.volley.ExecutorDelivery$ResponseDeliveryRunnable.run(ExecutorDelivery.java:99)
                                                     at android.os.Handler.handleCallback(Handler.java:739)
                                                     at android.os.Handler.dispatchMessage(Handler.java:95)
                                                     at android.os.Looper.loop(Looper.java:148)
                                                     at android.app.ActivityThread.main(ActivityThread.java:5417)
                                                     at java.lang.reflect.Method.invoke(Native Method)
                                                     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

### PR DESCRIPTION
**Pull request by maxme**
Fix #164 
Waiting for https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/7 to be merged

### CODE REVIEW

**Comment by aforcier** [comment]
Merged https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/7.

**Comment by tonyr59h** [comment]
https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/9 opened to correct some typing. This PR should wait on that one to be merged as well.

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/xmlrpc/XMLRPCUtils.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            Object value = map.get("value");
+            if (type.isInstance(value)) {
+                return (T) value;
+            }
+
+            if (type == String.class) {
+                return (T) MapUtils.getMapStr(map, "value");
+            } else if (type == Boolean.class) {
+                return (T) Boolean.valueOf(MapUtils.getMapBool(map, "value"));
+            } else if (type == Integer.class) {
```
Comment (tonyr59h) [inline fluxc/src/main/java/org/wordpress/android/fluxc/network/xmlrpc/XMLRPCUtils.java]: What about `Long.class` and `MapUtils.getMapLong`?

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/xmlrpc/XMLRPCUtils.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                return (T) Integer.valueOf(MapUtils.getMapInt(map, "value", defaultValue));
+            } else if (type == Float.class) {
+                return (T) Float.valueOf(MapUtils.getMapFloat(map, "value", defaultValue));
+            } else if (type == Double.class) {
+                return (T) Float.valueOf(MapUtils.getMapDouble(map, "value", defaultValue));
```
Comment (tonyr59h) [inline fluxc/src/main/java/org/wordpress/android/fluxc/network/xmlrpc/XMLRPCUtils.java]: Should be `return (T) Double.valueOf(MapUtils.getMapDouble(map, "value", defaultValue));`

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/xmlrpc/XMLRPCUtils.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            if (type.isInstance(value)) {
+                return (T) value;
+            }
+
+            if (type == String.class) {
+                return (T) MapUtils.getMapStr(map, "value");
+            } else if (type == Boolean.class) {
+                return (T) Boolean.valueOf(MapUtils.getMapBool(map, "value"));
+            } else if (type == Integer.class) {
+                return (T) Integer.valueOf(MapUtils.getMapInt(map, "value", defaultValue));
```
Comment (tonyr59h) [inline fluxc/src/main/java/org/wordpress/android/fluxc/network/xmlrpc/XMLRPCUtils.java]: I'm getting: `Wrong 3rd argument type. Found: 'T', required: 'int'` for `defaultValue` on this call plus the two below. We should probably unbox it into an explicitly typed variable.

**Comment by tonyr59h** [comment]
The crash is resolved, good to merge once the above comments are addressed 👍

**Comment by maxme** [comment]
Fixed and added tests. I requested `org.wordpress:utils` to be distributed via jcenter, I hope that will be accepted quickly.

**Comment by tonyr59h** [comment]
:shipit:

**Pull request merged**